api/handler/multipart_upload.go

```diff
@@ -426,7 +427,28 @@ func (h *handler) CompleteMultipartUploadHandler(w http.ResponseWriter, r *http.
 		}
 	}
 
-	if _, err = h.obj.DeleteObjects(r.Context(), bktInfo, []*layer.VersionedObject{{Name: initPart.Name}}); err != nil {
+	s := &layer.SendNotificationParams{
+		Event:   notifications.EventObjectCreatedCompleteMultipartUpload,
+		ObjInfo: objInfo,
+		BktInfo: bktInfo,
+		ReqInfo: reqInfo,
+	}
+	if err := h.obj.SendNotifications(r.Context(), s); err != nil {
+		h.log.Error("couldn't send notification: %w", zap.Error(err))
+	}
+
+	bktSettings, err := h.obj.GetBucketSettings(r.Context(), bktInfo)
+	if err != nil {
+		h.logAndSendError(w, "could not get bucket settings", reqInfo, err)
+		return
+	}
+
```

Comment:
Maybe we should use this settings below to determine if we need set `api.AmzVersionID` header (lines 462-466)